run.c

```diff
@@ -745,6 +744,16 @@ int main(int argc, char *argv[]) {
     Sampler sampler;
     build_sampler(&sampler, transformer.config.vocab_size);
 
+
+    // Check for sound parameters and fallback to defaults if needed.
+    // This needs to stay close to the generation code since, in the future,
+    // it will be in the "chat loop" (one may change params from on prompts to the other).
+    if (rng_seed <= 0) { rng_seed = (unsigned int)time(NULL);}
+    if (temperature < 0.0 || 1.0 < temperature) temperature = 1.0;
```

Comment:
Why is `temperature` clamped to 1.0, especially without any user-visible diagnostics? It also seems that maybe it's better to print error message for `temperature < 0.0`